tests/sysio.roa_tests.cpp

```diff
@@ -0,0 +1,302 @@
+#include <boost/test/unit_test.hpp>
+#include <string>
+#include <sysio/testing/tester.hpp>
+#include <sysio/chain/abi_serializer.hpp>
+#include "sysio.system_tester.hpp"
+
+#include <fc/variant_object.hpp>
+#include <type_traits>
+
+using namespace sysio::testing;
+using namespace sysio;
+using namespace sysio::chain;
+using namespace sysio::testing;
+using namespace fc;
+using namespace std;
+
+using mvo = fc::mutable_variant_object;
+
+constexpr account_name ROA = "sysio.roa"_n;
+constexpr uint64_t NETWORK_GEN = 0;
+
+class sysio_roa_tester : public tester {
+public:
+
+   sysio_roa_tester() {
+      produce_blocks( 2 );
+
+      create_accounts( { "alice"_n, "bob"_n, "carol"_n, "darcy"_n } );
+      produce_blocks( 2 );
+
+      set_code( ROA, contracts::roa_wasm() );
+      set_abi( ROA, contracts::roa_abi().data() );
+
+      produce_blocks();
+
+      const auto& accnt = control->db().get<account_object,by_name>( ROA );
+      abi_def abi;
+      BOOST_REQUIRE_EQUAL(abi_serializer::to_abi(accnt.abi, abi), true);
+      abi_ser.set_abi(abi, abi_serializer::create_yield_function(abi_serializer_max_time));
+   }
+
+   action_result push_action( const account_name& signer, const action_name &name, const variant_object &data ) {
+      string action_type_name = abi_ser.get_action_type(name);
+
+      action act;
+      act.account = ROA;
+      act.name    = name;
+      act.data    = abi_ser.variant_to_binary( action_type_name, data, abi_serializer::create_yield_function(abi_serializer_max_time) );
+
+      return base_tester::push_action( std::move(act), signer.to_uint64_t() );
+   }
+
+   transaction_trace_ptr push_paid_action( const account_name& signer, const action_name &name, const variant_object &data ) {
+      return base_tester::push_action( ROA, name,
+         vector<permission_level>{{signer, "active"_n},{signer, "sysio.payer"_n}},
+         data);
+   }
+
+   fc::variant get_nodeowner( account_name acc )
```

Comment:
Looks like this finds the nodeowner for "acc" if there is a nodeowner's table, otherwise it just returns "acc" with a tier3 indication, right? If so, or whatever it is, would be good to have a comment indicating that.

Reply:
oops, that `else` was debugging code that needed to be removed 😅